PGM/src/main/java/tc/oc/pgm/commands/MapCommands.java

```diff
@@ -41,46 +51,125 @@
 import tc.oc.pgm.rotation.RotationState;
 import tc.oc.pgm.teams.TeamFactory;
 
-public class MapCommands {
+public class MapCommands implements Commands {
+    private final UserFinder userFinder;
+    private final SyncExecutor syncExecutor;
+    private final IdentityProvider identityProvider;
+    private final ComponentRenderContext renderer;
+
+    @Inject
+    MapCommands(UserFinder userFinder, IdentityProvider identityProvider, SyncExecutor syncExecutor, ComponentRenderContext renderer) {
+        this.userFinder = userFinder;
+        this.identityProvider = identityProvider;
+        this.syncExecutor = syncExecutor;
+        this.renderer = renderer;
+    }
+
     @Command(
-        aliases = {"maplist", "maps", "ml"},
-        desc = "Shows the maps that are currently loaded",
-        usage = "[page]",
-        min = 0,
-        max = 1,
-        help = "Shows all the maps that are currently loaded including ones that are not in the rotation."
+            aliases = {"maplist", "maps", "ml"},
+            desc = "Shows the maps that are currently loaded",
+            usage = "[-a author] [-g gamemode] [page]",
+            min = 0,
+            max = 2,
+            flags = "a:g:",
+            help = "Shows all the maps that are currently loaded including ones that are not in the rotation."
     )
     @CommandPermissions("pgm.maplist")
-    public static void maplist(CommandContext args, final CommandSender sender) throws CommandException {
-        final Set<PGMMap> maps = ImmutableSortedSet.copyOf(new PGMMap.DisplayOrder(), PGM.getMatchManager().getMaps());
+    public List<String> maplist(final CommandContext args, final CommandSender sender) throws CommandException {
+        Identity senderIdentity = identityProvider.createIdentity(sender);
+        if (args.getFlag('a') != null && !args.getFlag('a').isEmpty() && args.getFlag('g') != null && !args.getFlag('g').isEmpty()) {
```

Comment:
this if/else if/else if/else condition can be simplified by parsing gamemode first, always (make "parseGamemode" return null if input is null) and only filter if/else if there's an author or not

Get the translatable component with a ? operator, or even create it before going into any conditionals